From d080001963a77ff89779be7e8c169ac13390c3bf Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Tue, 8 Oct 2024 13:08:40 +0530
Subject: [PATCH] ipmi: ipmb_dev_int: add quick fix for raw I2C type

Port patch to Ast2700 kernel source to ensure IPMB uses I2C
protocol over SMBus.

Ref: https://git.ami.com/core/ami-bmc/base-tech/os.linux.kernel.openbmc.linux/-/commit/e1b0bafb048cefdd4b2fb403512e3faae0a2a093

Tested and Verified:
1. Successfully able to get IPMB Request
root@intel-obmc:~# busctl call xyz.openbmc_project.Ipmi.Channel.Ipmb \
/xyz/openbmc_project/Ipmi/Channel/Ipmb org.openbmc.Ipmb \
sendRequest yyyyay 1 6 0 1 0
(iyyyyay) 0 7 0 1 0 15 32 129 13 6 2 191 0 0 0 2 2 70 22 3 0
root@intel-obmc:~#

Signed-off-by: vipinc <vipinc@ami.com>
---
 drivers/char/ipmi/ipmb_dev_int.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/char/ipmi/ipmb_dev_int.c b/drivers/char/ipmi/ipmb_dev_int.c
index 49100845fcb7..a2ee6ad58065 100644
--- a/drivers/char/ipmi/ipmb_dev_int.c
+++ b/drivers/char/ipmi/ipmb_dev_int.c
@@ -279,6 +279,7 @@ static int ipmb_slave_cb(struct i2c_client *client,
 		break;
 
 	case I2C_SLAVE_WRITE_RECEIVED:
+
 		if (ipmb_dev->msg_idx >= sizeof(struct ipmb_msg) - 1)
 			break;
 
@@ -327,8 +328,12 @@ static int ipmb_probe(struct i2c_client *client)
 	if (ret)
 		return ret;
 
+#if 1 /* FIXME: Quick fix. Need to add parsing code for ipmb_id instead */
+	ipmb_dev->is_i2c_protocol = true;
+#else
 	ipmb_dev->is_i2c_protocol
 		= device_property_read_bool(&client->dev, "i2c-protocol");
+#endif
 
 	ipmb_dev->client = client;
 	i2c_set_clientdata(client, ipmb_dev);
-- 
2.25.1

